src/dune_rules/coq/coq_rules.ml

```diff
@@ -331,17 +303,50 @@ let setup_coqdep_rule ~sctx ~dir ~loc ~theories_deps ~wrapper_name ~use_stdlib
     >>> Action_builder.(with_no_targets (goal source_rule))
     >>> Command.run ~dir:(Path.build dir) ~stdout_to coqdep file_flags)
 
-let deps_of ~dir ~use_stdlib ~wrapper_name coq_module =
-  let stdout_to = Coq_module.dep_file ~obj_dir:dir coq_module in
+module Dep_map = Stdune.Map.Make (Path)
+
+let coqdep_invalid phase line =
+  User_error.raise
+    [ Pp.textf "coqdep returned invalid output [phase: %s]" phase
+    ; Pp.verbatim line
+    ]
+
+let parse_line ~dir ~boot_type line =
+  match String.lsplit2 line ~on:':' with
+  | None -> coqdep_invalid "split" line
+  | Some (basename, deps) ->
+    let target = List.hd @@ String.extract_blank_separated_words basename in
+    (* let depname, ext = Filename.split_extension ff in *)
+    let target = Path.relative (Path.build dir) target in
```

Comment:
target needs to be handled better, for example to filter .vio files out, also to guard against permuations, the `List.hd` handling is a bit hacky.